### PR TITLE
[MOB-5757] messages auto fetch timer

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 Jira Ticket(s)

* [MOB-5757](https://iterable.atlassian.net/browse/MOB-5757)

## ✏️ Description

* implements timer for automatic message fetching

[MOB-5757]: https://iterable.atlassian.net/browse/MOB-5757?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swift-sdk/Internal/InternalIterableAPI.swift [hunk 1]
```diff
@@ -73,7 +73,9 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
     }()
     
     lazy var embeddedMessagingManager: IterableEmbeddedMessagingManagerProtocol = {
-        self.dependencyContainer.createEmbeddedMessagingManager()
+        self.dependencyContainer.createEmbeddedMessagingManager(config: self.config,
+                                                                apiClient: self.apiClient,
+                                                                dateProvider: self.dateProvider)
```
Comment (Ayyanchira) [inline swift-sdk/Internal/InternalIterableAPI.swift:78]: Curious why the constructor needs date provider? Is embdeddedManager going to use that inside for timecheck?

File: swift-sdk/Internal/EmbeddedMessagingManager.swift [hunk 2]
```diff
@@ -37,28 +43,88 @@ class EmbeddedMessagingManager: NSObject, IterableEmbeddedMessagingManagerProtoc
...
+                    ITBError()
+                })
+    }
+    
+    private func trackDeliveries(messages: [IterableEmbeddedMessage]) {
+        // TODO: track deliveries
     }
     
+    private func notifyUpdateDelegates(messages: [IterableEmbeddedMessage]) {
+        // TODO: filter `messages` by `placementId` and notify objects in `listeners` that have that placement ID
```
Comment (Ayyanchira) [inline swift-sdk/Internal/EmbeddedMessagingManager.swift:116]: Good! Got this one slipped on Android... Will add a TODO there as well 👍🏼

File: swift-sdk/Constants.swift [hunk 3]
```diff
@@ -204,6 +204,11 @@ enum JsonKey {
         static let userInterfaceIdiom = "userInterfaceIdiom"
     }
     
+    enum Embedded {
+        static let packageName = "packageName"
+        static let sdkVersion = "SDKVersion"
```
Comment (Ayyanchira) [inline swift-sdk/Constants.swift:209]: Why within Embedded?

**Review by Ayyanchira — APPROVED** [review]
Looking good. Made some comments. Approving as its going in the embedded-messaging

File: swift-sdk/Constants.swift (hunk 3 above)
Comment (evangreer91) [reply swift-sdk/Constants.swift:209]: This is similar to the way in-app is defined.

File: swift-sdk/Internal/ApiClient.swift [hunk 4]
```diff
@@ -218,7 +218,7 @@ extension ApiClient: ApiClientProtocol {
     }
     
     // MARK: - Embedded Messaging
-    func getEmbeddedMessages() -> Pending<SendRequestValue, SendRequestError> {
+    func getEmbeddedMessages() -> Pending<[IterableEmbeddedMessage], SendRequestError> {
         let result = createRequestCreator().flatMap { $0.createGetEmbeddedMessagesRequest() }
```
Comment (evangreer91) [inline swift-sdk/Internal/ApiClient.swift:222]: Why is there a difference in the SendRequestValue for in-app vs. embedded?

File: swift-sdk/Internal/InternalIterableAPI.swift (hunk 1 above)
Comment (roninopf) [reply swift-sdk/Internal/InternalIterableAPI.swift:78]: Yep, we also pass it in for testing (for mocking time/date).

File: swift-sdk/Constants.swift (hunk 3 above)
Comment (roninopf) [reply swift-sdk/Constants.swift:209]: I wanted to give it its own namespace within the constants. Previously it was using the in-app's constants.

File: swift-sdk/Internal/ApiClient.swift (hunk 4 above)
Comment (roninopf) [reply swift-sdk/Internal/ApiClient.swift:222]: For embedded messaging, it'll automatically serialize it to that format, where as `SendRequestValue` is a generic return value.

**Pull request merged**